src/clj/fluree/db/json_ld/migrate/sid.cljc

```diff
@@ -21,109 +24,139 @@
+                                 (set-namespaces ns-mapping))
+          committed-db       (<? (jld-ledger/commit! ledger staged-db
+                                                     {:time (get-first-value commit const/iri-time)}))]
+      (if (async-db/db? committed-db)
+        (<? (async-db/deref-async committed-db))
+        committed-db))))
 
-;;           commit-id          (:id commit-metadata)
-;;           commit-sid         (iri/encode-iri db* commit-id)
-;;           [prev-commit _]    (some->> previous :address (reify/read-commit conn) <?)
-;;           db-sid             (iri/encode-iri db* (:id data))
-;;           metadata-flakes    (commit-data/commit-metadata-flakes commit-metadata
-;;                                                                  t-new commit-sid db-sid)
-;;           previous-id        (when prev-commit (:id prev-commit))
-;;           prev-commit-flakes (when previous-id
-;;                                (commit-data/prev-commit-flakes db* t-new commit-sid
-;;                                                                previous-id))
-;;           prev-data-id       (get-first-id prev-commit const/iri-data)
-;;           prev-db-flakes     (when prev-data-id
-;;                                (commit-data/prev-data-flakes db* db-sid t-new
-;;                                                              prev-data-id))
-;;           issuer-flakes      (when-let [issuer-iri (:id issuer)]
-;;                                (commit-data/issuer-flakes db* t-new commit-sid issuer-iri))
-;;           message-flakes     (when message
-;;                                (commit-data/message-flakes t-new commit-sid message))
-;;           all-flakes         (-> db*
-;;                                  (get-in [:novelty :spot])
-;;                                  empty
-;;                                  (into metadata-flakes)
-;;                                  (into retracted-flakes)
-;;                                  (into asserted-flakes)
-;;                                  (cond->
-;;                                      prev-commit-flakes (into prev-commit-flakes)
-;;                                      prev-db-flakes (into prev-db-flakes)
-;;                                      issuer-flakes  (into issuer-flakes)
-;;                                      message-flakes (into message-flakes)))]
-;;       (when (empty? all-flakes)
-;;         (reify/commit-error "Commit has neither assertions or retractions!"
-;;                       commit-metadata))
-;;       (-> db*
-;;           (reify/merge-flakes t-new all-flakes)
-;;           (assoc :previous (:commit db*))
-;;           (assoc :commit commit-metadata)))))
+(defn migrate-commits
+  "Reduce over each commmit and integrate its data into the ledger's db."
+  [ledger branch tuples-chans]
+  (go-try
+    (loop [[[commit-tuple ch] & r] tuples-chans
+           db (<? (async-db/deref-async (jld-ledger/current-db ledger)))]
```

Comment:
it's possible that current db is a FlakeDB. I think you're going to have to check what kind of db it is before trying to deref it.

Reply:
This ledger is a brand new one returned from a call to `jld-ledger/create*` with no data transacted into it, so I think it will always be an AsyncDB the first time you call `current-db` on it, but I'll add a guard just in case.